proposals/mesh-detection.html

```diff
@@ -0,0 +1,499 @@
+
+        const light = new THREE.HemisphereLight(0xffffff, 0xbbbbff, 1);
+        light.position.set(0.5, 1, 0.25);
+        scene.add(light);
+
+        renderer = new THREE.WebGLRenderer({ antialias: true, alpha: true });
+        renderer.setPixelRatio(window.devicePixelRatio);
+        renderer.setSize(window.innerWidth, window.innerHeight);
+        renderer.xr.enabled = true;
+        renderer.xr.setFoveation(0);
+        renderer.autoClear = false;
+        container.appendChild(renderer.domElement);
+
+        xrButton = new WebXRButton({
+          onRequestSession: onRequestSession,
+          onEndSession: onEndSession,
+          textEnterXRTitle: "START AR",
+          textXRNotFoundTitle: "AR NOT FOUND",
+          textExitXRTitle: "EXIT AR",
+        });
+
+        document.querySelector('header').appendChild(xrButton.domElement);
+
+        if (navigator.xr) {
+          navigator.xr.isSessionSupported('immersive-ar')
+                      .then((supported) => {
+            xrButton.enabled = supported;
+          });
+        }
+
+
+				controller1 = renderer.xr.getController(0);
+				controller1.addEventListener('selectstart', onSelectStart);
+				controller1.addEventListener('selectend', onSelectEnd);
+				controller1.addEventListener('connected', function (event) {
+
+					this.add(buildController(event.data));
+
+				});
+				controller1.addEventListener('disconnected', function () {
+
+					this.remove(this.children[ 0 ]);
+
+				});
+				scene.add(controller1);
+
+				controller2 = renderer.xr.getController(1);
+				controller2.addEventListener('selectstart', onSelectStart);
+				controller2.addEventListener('selectend', onSelectEnd);
+				controller2.addEventListener('connected', function (event) {
+
+					this.add(buildController(event.data));
+
+				});
+        scene.add(controller2);
+
+				const controllerModelFactory = new XRControllerModelFactory();
+
+				controllerGrip1 = renderer.xr.getControllerGrip(0);
+				controllerGrip1.add(controllerModelFactory.createControllerModel(controllerGrip1));
+				scene.add(controllerGrip1);
+
+				controllerGrip2 = renderer.xr.getControllerGrip(1);
+				controllerGrip2.add(controllerModelFactory.createControllerModel(controllerGrip2));
+				scene.add(controllerGrip2);
+
+				controller2.addEventListener('disconnected', function () {
+
+					this.remove(this.children[ 0 ]);
+
+				});
+				scene.add(controller2);
+
+        reticle = new THREE.Mesh(
+          new THREE.SphereGeometry(0.15,32,16), //RingGeometry(0.15, 0.2, 32).rotateX(-Math.PI / 2),
+          new THREE.MeshBasicMaterial({color: 0xff0000})
+      );
+        reticle.matrixAutoUpdate = true;
+        reticle.visible = false;
+        scene.add(reticle);
+
+        const createMeshMaterial = (params) => 
+          new THREE.MeshBasicMaterial(Object.assign(params, {
+            opacity: 0.15,
+            transparent: true,
+          }));
+
+        if (useMeshOrigin.checked) {        
```

Comment:
Is this backwards? Feels like you want the meshes to be additively blended if the origin wireframes are being shown and opaque otherwise? (Haven't had a chance to actually try this myself, so I could be wrong  about the intent)

Reply:
I renamed the variable so it's more clear what's happening.
Basically if the mesh isn't visible, it will knock out content that is behind it